src/main/java/amara/command/AddDeadlineCommand.java

```diff
@@ -0,0 +1,39 @@
+package amara.command;
+
+import java.util.ArrayList;
+
+import amara.storage.Storage;
+import amara.task.Deadline;
+import amara.task.Task;
+import amara.ui.Ui;
+
+/**
+ * A {@link Command} implementation that adds a {@link Deadline} task
+ * to a given {@code ArrayList<Task>}.
+ * <p>
+ * The given {@link Deadline} task is  added to the provided {@code ArrayList<Task>}.
+ * </p>
+ */
+public class AddDeadlineCommand extends Command {
+    private static final String MESSAGE = "Got it. I've added this task:\n"
+            + "  %s\nNow you have %d tasks in the list.";
+    private Deadline deadline;
+
+    public AddDeadlineCommand(Deadline deadline) {
+        this.deadline = deadline;
+    }
+
+    /**
+     * Executes the {@code Command}.
+     * @param tasks List of tasks.
+     * @param ui UI handler.
+     * @param storage To store the given List of tasks.
+     */
```

Comment:
```suggestion
    /**
     * Adds the Deadline to a given ArrayList<Task>, forms a message for a UI handler to display
     * 
     * @param tasks List of tasks.
     * @param ui UI handler.
     * @param storage To store the given List of tasks.
     * @return Message given to UI handler
     */
```
Coding standard stuff: A line is needed between the description and the parameters. I think all your method JavaDocs have this problem.
Also, the description is quite vague, perhaps you can consider elaborating a little more, something like the example given.